networkx/classes/coreviews.py

```diff
@@ -138,7 +138,7 @@ def __len__(self):
         return len(self._succ.keys() | self._pred.keys())
 
     def __iter__(self):
-        return iter(set(self._succ.keys()) | set(self._pred.keys()))
+        return iter(set(self._succ) | set(self._pred))
```

Comment:
You should probably also try: 
`self._succ.keys() | self._pred.keys()` 
which would not create any new data objects. For large cases it should be faster because of that.